### DISCUSSION TITLE
Support for native container queries?

### DISCUSSION THREAD

**Discussion by jbwharris** [postcss/postcss#1852] [General]
I am really new to PostCSS and am wondering how to get container queries functioning. The issue I'm encountering is my ﹫container declarations are just skipped at compile. I added container-name and container-type to be ignored properties in my linting, as well as added container to the ignore rules. I've tested the queries themselves work, but can't get them to be included in my compile step. I've seen [this plugin](https://github.com/ZeeCoder/container-query), but it hasn't been updated since 2019, so was unsure if that was the right path considering I'm less looking for a polyfill than native support. Can anyone shed some light on what I might be messing up?

**Comment by ai** [comment]
I never use it to help you here. But note that this polyfill requires also JS file on client-side (not only PostCSS plugin).

Honestly, there is no good container queries polyfill.

**Reply by jbwharris** [reply]
Hmm, that does seem disappointing, though a bit understandable knowing how nuanced the functionality is. I find I'm less concerned about a polyfill for the case I'm trying to implement it, I just want it to work in it's native form without requiring me to add a separate CSS file cludge for it. I can get it to work with a <style> tag in the head, but it just refuses to acknowledge it in the compile.

**Comment by VityaSchel** [comment]
Now that it is baseline available since 2023 can this be added? It wouldn't be so problematic if it wasn't for the warnings that postcss emit for those "unknown properties"

**Reply by ai** [reply]
PistCSS itself has no `unknown properties` error and should parse CSS with it without a problem.

Seems like you have another tool which generates this error.
